redfish-core/lib/log_services.hpp

```diff
@@ -4082,6 +4070,30 @@ inline void handleLogServicesDumpEntryComputerSystemDelete(
     deleteDumpEntry(asyncResp, dumpEntryId, "System");
 }
 
+inline void handleLogServicesDumpEntryComputerSystemDownloadGet(
+    crow::App& app, const std::string& dumpType, const crow::Request& req,
+    const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+    const std::string& /*dumpId*/)
+{
+    if (!redfish::setUpRedfishRoute(app, req, asyncResp))
+    {
+        return;
+    }
+    std::string url(req.target());
```

Comment:
look like this code can be taken out in to a function that returns optional value containing the url segment between Entries and attachement

Reply:
Its just at this place we are having this requirement. 
I guess this should still be fine.

Do you still think if another function is required?

Reply:
but it is doing potentially reusable function of extracting the needed url segment from url. This has nothing to do actual function. So separate the concerns, which will improve readability and future reusability.